server/src/server.rs

```diff
@@ -175,11 +182,12 @@ where
 			}
 		}
 
-		// FuturesUnordered won't poll anything until this line but because the
-		// tasks are spawned (so that they can progress independently)
-		// then this just makes sure that all tasks are completed before
-		// returning from this function.
-		while connections.next().await.is_some() {}
+		// Drop the last notifier
+		drop(drop_on_completion);
+
+		// Wait for completion of connection processing tasks.
+		// We will stop waiting here once all senders are dropped
+		let _ = process_connection_awaiter.recv().await;
```

Comment:
Can you make a comment that no message should be sent on this channel and once this returns all connections have been gracefully shutdown?

Reply:
@niklasad1 Added

Reply:
@niklasad1 added